lib/God.js

```diff
@@ -176,15 +176,29 @@ God.executeApp = function executeApp(env, cb) {
         God.handleExit(clu, code || 0, signal || 'SIGINT');
       });
 
-      clu.once('online', function cluOnline() {
-        console.log('App name:%s id:%s online', clu.pm2_env.name, clu.pm2_env.pm_id);
-        clu.pm2_env.status = cst.ONLINE_STATUS;
+      var readyCb = function ready() {
         if (clu.pm2_env.vizion !== false && clu.pm2_env.vizion != "false")
           God.finalizeProcedure(clu);
         else
           God.notify('online', clu);
+        
+        clu.pm2_env.status = cst.ONLINE_STATUS;
+        console.log('App name:%s id:%s online', clu.pm2_env.name, clu.pm2_env.pm_id);
         if (cb) cb(null, clu);
-      });
+      }
+
+      if (clu.pm2_env.wait_ready) {
+        var listener = function (packet) {
+          if (packet.raw === 'ready' && packet.process.name === clu.pm2_env.name &&
+                packet.process.pm_id === clu.pm2_env.pm_id) {
+            God.bus.removeListener('ready', listener)
```

Comment:
We call  `removeListener` on `ready` but we listen on `process:msg`. Can you explain to me how this works?